### ISSUE TITLE
[Training] Add support for Dropout in Forge

### ISSUE THREAD

**Issue by umalesTT** [tenstorrent/tt-mlir#4854]
Dropout is not supported in our Forge compiler, which consumes both Jax and Torch frontends. Compilation for both frontends fails during stablehlo->ttir legalization pass.
For Jax unsupported op is [stablehlo.bitcast_convert](https://github.com/tenstorrent/tt-mlir/issues/979) and for Torch it is [stablehlo.rng_bit_generator](https://github.com/tenstorrent/tt-mlir/issues/4793).

Both of these ops are currently unsupported in metal so one way forward is to fuse dropout operations in stablehlo for both frontends, therefore circumventing those 2 unsupported ops and lowering the fused operation onto [ttnn.experimental.dropout](https://docs.tenstorrent.com/tt-metal/v0.55.0/ttnn/ttnn/api/ttnn.experimental.dropout.html) - which, irregardless of the mentioned ops, should be the way to go efficiency-wise.

So the general workflow would be:

1. Add fusing patterns for dropout in stablehlo for both frontends. In offline talks with @azecevicTT we concluded that for Jax dropout should be treated as composite op and monkey-patched in Jax, much like the [gelu case](https://github.com/tenstorrent/tt-xla/pull/785), while for Torch regular fusing pattern is best until we support composite ops as they are in Jax. Because of this differences, these will be 2 separate issues.
2. Add ```ttir.dropout``` op which will be compatible with both frontend paths from 1.
3. Lower ```ttir.dropout``` to ```ttnn.experimental.dropout```

Concern that remains here is that the experimental metal op should have all [rng algorithms of interest](https://openxla.org/stablehlo/spec#rng_bit_generator) supported as otherwise we won't get same results as CPU.

**Comment by umalesTT** [comment]
After some additional testing and offline discussions, plan for Jax is modified (plan for Torch stays the same).
When dropout in Jax is fused as in initial idea, some occurrences of stablehlo.bitcast_convert still emerge in training and the next largest function we may fuse in order to eliminate mentioned op is jax.random.uniform (https://github.com/tenstorrent/tt-mlir/issues/5066).

Once the stablehlo.bitcast_convert is supported in our stack we will migrate to initial plan for Jax.

**Comment by sdjordjevicTT** [comment]
Dropout op is now fully supported in ttmlir with this PR:
https://github.com/tenstorrent/tt-mlir/pull/6656
